common/windows-app-bootstrap.cpp

```diff
@@ -10,11 +10,141 @@
 #include <vector>
 #include <algorithm>
 #include <iterator>
+#include <sstream>
+#include <iostream>
+#include <csignal>
 
+#include "os.h"
 #include "metadata-helper.h"
+#include "rendering.h"
+#include <delayimp.h>
 
+// Use OS hook to modify message box behaviour
+// This lets us implement "report" functionality if Viewer is crashing
+HHOOK hhk;
+// When launching messagebox, replace standard Cancel button with "report"
```

Comment:
Does this mean that, when the user hits the 'X' on the top-right of the box, 'Report' will happen?

Reply:
No, if you click X (and everything closes-up correctly) return code from main will be 0, skipping the message box.

Reply:
I meant if you click 'x' on the message box... user will not expect that to open a web page

Reply:
Web page is only opened when the user clicks on the "Cancel" button (tested)